### PR TITLE
Scripts/SQL/Core: Implement Bar- spell magic defense bonus

### PR DESCRIPTION
**Pull request by Hozu**
Bar- spells all set the subpower to 1 for whatever reason, and a search through all of the files that contain getSubPower() had nothing in regards to bar- spells. So I'm using it to hold the bonus magic defense bonus for the element, and I removed it for the status resistance spells.

Source: https://www.bg-wiki.com/bg/Category:Barspell#Increasing_Elemental_Barspell_Potency

### CODE REVIEW

**Comment by Hozu** [comment]
Also changed the description of the force day/weather bonus mod to be what the code actually checks for. I should have wrote it like that to begin with.

**Comment by Hozu** [comment]
Added a really small change - noticed that there was a spell:getElement() despite there being a previously defined variable containing that value, so I replaced it.

File: scripts/globals/monstertpmoves.lua [hunk 1]
```diff
@@ -396,7 +400,12 @@ function mobAddBonuses(caster, spell, target, dmg, ele)
 
     dmg = math.floor(dmg * burst);
 
-    mab = (100 + caster:getMod(MOD_MATT)) / (100 + target:getMod(MOD_MDEF)) ;
+    local mdefBarBonus = 0;
+    if (ele > 0 and ele <= 6 and target:hasStatusEffect(barSpells[ele])) then -- bar- spell magic defense bonus, held in subid
+        mdefBarBonus = target:getStatusEffect(barSpells[ele]):getSubType();
+		print("test")
```
Comment (xdemolish) [inline scripts/globals/monstertpmoves.lua]: sneaky print

Comment (bendangelo) [inline scripts/globals/monstertpmoves.lua]: I thought bar spells don't give magic defense bonus but only a higher resist rate.

Comment (Hozu) [inline scripts/globals/monstertpmoves.lua]: Whoops I forgot to remove that. As for bar spells: https://www.bg-wiki.com/bg/Category:Barspell#Increasing_Elemental_Barspell_Potency

**Comment by Hozu** [comment]
Removed the print that I forgot to remove. Also included the source in the commit message.

**Comment by teschnei** [comment]
can you switch the barspells to use subpower instead of subid? (or even tier)
subid is kind of reserved and the core does unexpected things with it if you don't use it right

**Comment by Hozu** [comment]
I asked about subpower before, since it's always being set to a specific value (1), you said to use something else. I'll switch it to tier when I get the chance. Or, why do they always have subpower 1?

**Comment by teschnei** [comment]
I have no idea. check the barspell effect script, but there might be other
places

**Comment by Hozu** [comment]
I searched for getSubPower and looked through every file that it was in, and none of them was looking at the barspell effect. Strange... I suppose it was used for something in the past, but that got removed. I'll change it over to subpower then.

**Comment by Hozu** [comment]
Yeah I'm going to use subpower for this mod. Nothing that getSubPower is involved in is getting it from a bar- spell. It must have been from something that got removed at some point.

**Pull request merged**